tracer/test/Datadog.Trace.Tests/Agent/MessagePack/SpanMessagePackFormatterTests.cs

```diff
@@ -111,6 +113,83 @@ public void SerializeSpans()
         }
     }
 
+    [Fact]
+    public void SpanLink_Tag_Serialization()
+    {
+        var formatter = SpanFormatterResolver.Instance.GetFormatter<TraceChunkModel>();
+
+        var parentContext = new SpanContext(new TraceId(0, 1), 2, (int)SamplingPriority.UserKeep, "ServiceName1", "origin1");
+
+        var spans = new[]
+        {
+            new Span(parentContext, DateTimeOffset.UtcNow),
+            new Span(new SpanContext(parentContext, new TraceContext(Mock.Of<IDatadogTracer>()), "ServiceName1"), DateTimeOffset.UtcNow),
+            new Span(new SpanContext(new TraceId(0, 5), 6, (int)SamplingPriority.UserKeep, "ServiceName3", "origin3"), DateTimeOffset.UtcNow),
+        };
+        var attributesToAdd = new List<KeyValuePair<string, object>>
+        {
+            new KeyValuePair<string, object>("link.name", "manually_linking"),
+            new KeyValuePair<string, object>("pair", false),
+            new KeyValuePair<string, object>("arbitrary", 56709)
+        };
+        spans[0].AddSpanLink(spans[1], attributesToAdd);
+        spans[1].AddSpanLink(spans[2]).AddAttribute("chaining", "adding attribute").AddAttribute("chaining2x", "we're cooking");
+        spans[1].AddSpanLink(spans[0]);
+
+        foreach (var span in spans)
+        {
+            span.SetDuration(TimeSpan.FromSeconds(1));
+        }
+
+        var traceChunk = new TraceChunkModel(new(spans));
+
+        byte[] bytes = [];
+
+        var length = formatter.Serialize(ref bytes, 0, traceChunk, SpanFormatterResolver.Instance);
+        var result = global::MessagePack.MessagePackSerializer.Deserialize<MockSpan[]>(new ArraySegment<byte>(bytes, 0, length));
+
+        for (int i = 0; i < result.Length; i++)
+        {
+            var expected = spans[i];
+            var actual = result[i];
+
+            if (expected.SpanLinks is not null)
+            {
+                for (int j = 0; j < expected.SpanLinks.Count; j++)
+                {
+                    var expectedSpanlink = expected.SpanLinks[j];
+                    var actualSpanLink = actual.SpanLinks[j];
+                    actualSpanLink.TraceIdHigh.Should().Be(expectedSpanlink.Context.TraceId128.Upper);
+                    actualSpanLink.TraceIdLow.Should().Be(expectedSpanlink.Context.TraceId128.Lower);
+                    actualSpanLink.SpanId.Should().Be(expectedSpanlink.Context.SpanId);
+                    var expectedTraceState = W3CTraceContextPropagator.CreateTraceStateHeader(expectedSpanlink.Context);
+                    if (string.IsNullOrEmpty(expectedTraceState))
```

Comment:
Wondering if this `if` is supposed to be negated as it seems like it'd only be asserting on `null`/`empty` at the moment

```suggestion
                    if (!string.IsNullOrEmpty(expectedTraceState))
```

Or maybe the `if` could just be removed entirely and could just do `actualSpanLink.TraceState.Should().Be(expectedTraceState);` for all?